resources/experimental-feature.md

```diff
@@ -0,0 +1,89 @@
+  - The discussion should not stay in a stale state. Users need to understand they can interact in the discussion and get quick answers. If needed, the discussion will be updated regularly.
+- Using the Meilisearch CLI: the `--help` command in the Meilisearch binary will redirect to the related GitHub discussion and warn about the unstable state of the features.
+- In the documentation: a small dedicated page should be done about the purpose of the experimental feature and redirect to the related GitHub discussion.
+
+## Some warnings about the usage
+
+- Experimental features are not recommended to be used in production.
+- For some provided experimental features, re-indexing will be needed. For this, more than only restarting the search engine, using a dump will be needed to enforce re-indexing.
+This means, in some situations, using a dump might be needed to activate an experimental feature but also to deactivate it. The users will be clearly informed in the related discussion how to activate the feature
+
+> ⚠️ When deactivating this kind of experimental feature, since our process is not mature yet, users can have some issues with their DB, even when using a dump.<br>
+> We recommend users always save their data (with snapshots and/or dumps) before starting to activate experimental features. Of course, the safest method is to use a script to feed Meilisearch.
+
+## Technical explanations
+
+### Why does Meilisearch need to be restarted (by using the `experimental` CLI flag)?
+
+Meilisearch uses LMDB to store the data, not only user documents but also a lot of internal data, like Meilisearch tasks. With LMDB, if we want to add or remove data structures to store this data, we need to close the LMDB environment and re-open it, so to restart Meilisearch.
+
+It means if an experimental feature implementation involves a modification of the data structures, users must restart Meilisearch; we cannot make it enable/disable from the HTTP routes.
+
+Unfortunately, this might concern most of the future features. That’s why we chose, for the moment, only to suggest flag CLI in order to enable/disable experimental features. However, this can be changed in the future or adapted to the situation.
+
+### Why could we need to use a dump?
+
+Since we need to restart Meilisearch, and changes happened on the DB side, an issue could happen when Meilisearch tries to read the “old” data.ms.
+
+Indeed, Meilisearch could try to read the old DB, but does not succeed to do it (crash), or do it badly (inconsistent results). That’s why we ask for dumps between 2 minor versions (there are often a lot of changes in the DB), and that’s why users sometimes will need to use a dump when passing from stable to experimental. The contrary (from experimental to stable) is riskier, even when using a dump, since we are not mature enough yet, but we will be safer and safer in the future.
```

Comment:
```suggestion
### Why will some features require migrating data with dumps?

Under some circumstances, Meilisearch might have issues when reading a database generated by a different Meilisearch release. This might cause an instance to crash or work with faulty data.

This is already a possibility when migrating between minor Meilisearch versions, and is more likely to happen when activating a new experimental feature. The opposite operation—migrating a database with experimental features activated to a database where those features are not active—is currently riskier. As we develop and improve the development of experimental features, this procedure will become safer and more reliable.
```